lib/cancan/model_additions.rb

```diff
@@ -22,7 +22,7 @@ module ClassMethods
       # Here only the articles which the user can update are returned.
       def accessible_by(ability, action = :index, strategy: CanCan.accessible_by_strategy)
         CanCan.with_accessible_by_strategy(strategy) do
-          ability.model_adapter(self, action).database_records
+          ability.model_adapter(all, action).database_records
```

Comment:
What if this is already being called from a scope?

```ruby
User.active.accessible_by(ability, :read)
```

I think that switching this to `all` causes `accessible_by` to return a new scope ignore all previous filters

Reply:
The opposite is true: current behavior is to ignore all previous filters, and this change will take them into account. 

Try in any project you have: 

`User.active.all.count` vs
`User.active.klass.all.count` - using `klass` (or `self` in the above code) will unscope the relation, whereas `all` will not.

Reply:
Gotcha, I thought `self` in that context was a AR relation.